File: ambeth/Ambeth.Service/ambeth/service/DefaultServiceUrlProvider.cs
using De.Osthus.Ambeth.Config;
using De.Osthus.Ambeth.Ioc;
using System;

namespace De.Osthus.Ambeth.Service
{
    public class DefaultServiceUrlProvider : IServiceUrlProvider, IOfflineListenerExtendable
    {
        public bool IsOffline
        {
            get
            {
                return false;
            }
            set
            {
                throw new NotSupportedException("This " + typeof(IServiceUrlProvider).Name + " does not support this operation");
            }
        }

        [Property(ServiceConfigurationConstants.ServiceProtocol, DefaultValue = "http")]
        public String OnlineServiceProtocol { protected get; set; }

        [Property(ServiceConfigurationConstants.ServiceHostName, DefaultValue = "localhost")]
        public String OnlineServiceHostName { protected get; set; }

        [Property(ServiceConfigurationConstants.ServiceHostPort, DefaultValue = "8000")]
        public uint OnlineServiceHostPort { protected get; set; }
        
        public void LockForRestart(bool offlineAfterRestart)
        {
            throw new NotSupportedException("This " + typeof(IServiceUrlProvider).Name + " does not support this operation");
        }

        public String GetServiceURL(Type serviceInterface, String serviceName)
        {
            return OnlineServiceProtocol + "://" + OnlineServiceHostName + ":" + OnlineServiceHostPort + "/" + serviceName + "/";
        }

        public void AddOfflineListener(IOfflineListener offlineListener)
        {
            // Intended NoOp!
        }

        public void RemoveOfflineListener(IOfflineListener offlineListener)
        {
            // Intended NoOp!
        }
    }
}

File: jambeth/jambeth-service/src/main/java/de/osthus/ambeth/service/DefaultServiceUrlProvider.java
package de.osthus.ambeth.service;

import de.osthus.ambeth.config.Property;
import de.osthus.ambeth.config.ServiceConfigurationConstants;
import de.osthus.ambeth.ioc.annotation.Autowired;
import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.objectcollector.IThreadLocalObjectCollector;
import de.osthus.ambeth.util.StringBuilderUtil;

public class DefaultServiceUrlProvider implements IServiceUrlProvider, IOfflineListenerExtendable
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	@Autowired
	protected IThreadLocalObjectCollector objectCollector;

	@Property(name = ServiceConfigurationConstants.ServiceBaseUrl, defaultValue = "http://localhost:8000")
	protected String serviceBaseUrl;

	@Override
	public String getServiceURL(Class<?> serviceInterface, String serviceName)
	{
		return StringBuilderUtil.concat(objectCollector.getCurrent(), serviceBaseUrl, "/", serviceName, "/");
	}

	@Override
	public boolean isOffline()
	{
		return false;
	}

	@Override
	public void setOffline(boolean isOffline)
	{
		throw new UnsupportedOperationException("This " + IServiceUrlProvider.class.getSimpleName() + " does not support this operation");
	}

	@Override
	public void lockForRestart(boolean offlineAfterRestart)
	{
		throw new UnsupportedOperationException("This " + IServiceUrlProvider.class.getSimpleName() + " does not support this operation");
	}

	@Override
	public void addOfflineListener(IOfflineListener offlineListener)
	{
		// Intended NoOp!
	}

	@Override
	public void removeOfflineListener(IOfflineListener offlineListener)
	{
		// Intended NoOp!
	}
}
